internal/storage/conn.go

```diff
@@ -59,7 +62,19 @@ func InitConn(params ConnParams) *Connection {
 func (c *Connection) Connect() error {
 	dsn := MakeDSN(c.params)
 	c.logger.LogF(monitor.F{"dsn": dsn}).Info("connecting to the DB")
-	db, err := sqlx.Connect(c.dialect, dsn)
+	var err error
+	var secondsToWait int
+	var db *sqlx.DB
+	for i := 0; i < maxDBConnectAttempts; i++ {
+		db, err = sqlx.Connect(c.dialect, dsn)
+		if err == nil {
+			break
+		}
+		secondsToWait = secondsToWait + i + 1
```

Comment:
This is a neat idea. But having to wait up to 220 seconds for the database will create problems with k8s deployments where we don't want app in an uncertain state for that long. I think 30 seconds is a reasonable baseline.